docs/operations.md

```diff
@@ -0,0 +1,141 @@
+
+To add these packages to the current cluster, run the same command as for [Reconfiguring Slurm](#Reconfiguring-Slurm). TODO: describe what's required to add these to site-specific images.
+
+If additional repositories are required, these could be added/enabled as necessary in a play added to `environments/$SITE_ENV/hooks/{pre,post}.yml` as appropriate. Note such a plat should NOT exclude the builder group, so that the repositories are also added to built images. There are various Ansible modules which might be useful for this:
+    - `ansible.builtin.yum_repository`: Add a repo from an URL providing a 'repodata' directory.
+    - `ansible.builtin.rpm_key` : Add a GPG key to the RPM database.
+    - `ansible.builtin.get_url`: Can be used to install a repofile directly from an URL (e.g. https://turbovnc.org/pmwiki/uploads/Downloads/TurboVNC.repo)
+    - `ansible.builtin.dnf`: Can be used to install 'release packages' providing repos, e.g. `epel-release`, `ohpc-release`.
+
+The packages to be installed from that repo could also be defined in that play. Note using the `dnf` module with a list for its `name` parameter is more efficient and allows better dependency resolution than calling the module in a loop.
+
+
+Adding these repos/packages to the cluster/image would then require running:
+
+    ansible-playbook environments/$SITE_ENV/hooks/{pre,post}.yml
+
+as appropriate.
+
+TODO: improve description about adding these to extra images.
+
+
+# Reconfiguring Slurm
+
+At a minimum run:
+
+    ansible-playbook ansible/slurm.yml --tags openhpc
+
+
+**NB:** This will restart all daemons if the `slurm.conf` has any changes, even if technically only a `scontrol reconfigure` is required.
+
+
+# Running the MPI Test Suite
+
+See [ansible/roles/hpctests/README.md](ansible/roles/hpctests/README.md) for a description of these. They can be run using
+
+    ansible-playbook ansible/adhoc/hpctests.yml
+
+Note that:
+- The above role provides variables to select specific partitions, nodes and interfaces which may be required. If not set in inventory, these can be passed as extravars:
+
+        ansible-playbook ansible/adhoc/hpctests.yml -e hpctests_myvar=foo
+- The HPL-based test is only resonably optimised on Intel processors due the libaries and default parallelisation scheme used. For AMD processors it is recommended this
+is skipped using:
+
+        ansible-playbook ansible/adhoc/hpctests.yml --skip-tags hpl-solo.
+
+Review any [site-specific documentation](site/README.md) for more details.
+
+# Running CUDA Tests
+This uses the [cuda-samples](https://github.com/NVIDIA/cuda-samples/) utilities "deviceQuery" and "bandwidthTest" to test GPU functionality. It automatically runs on any
+host in the `cuda` inventory group:
+
+    ansible-playbook ansible/adhoc/cudatests.yml
+
+**NB:** This test is not launched through Slurm, so confirm nodes are free/out of service or use `--limit` appropriately.
+
+# Ad-hoc Commands
```

Comment:
I think it is worth mentioning the ability to run arbitrary shell commands against ansible groups here i.e. `ansible <group/host> -m shell -a "<shell-command>" (optional: --become)`. I know it's just standard ansible, but at least I wasn't aware of how useful it was until relatively recently.

```suggestion
# Ad-hoc Commands and Playbooks
```